torch_xla/csrc/tensor_util.cpp

```diff
@@ -35,6 +35,34 @@ struct DataAsync {
   std::vector<xla::util::ExceptionCleanup> handle_unlockers;
 };
 
+void PopulateTensorBuffer(const at::Tensor& tensor,
```

Comment:
What's this?

Reply:
Okay, why not move PopulateTensorBuffer directly here?

Reply:
There are a few related functions to PopulateTensorBuffer that it makes sense to stay near. What do you think about just moving the TransferToServer wrappers down in the file instead? I don't think there's any reason that `TransferToServerAsync` has to be at the top.

Reply:
Updated the PR with the changes discussed